evennia/prototypes/tests.py

```diff
@@ -312,11 +344,24 @@ def test_protfuncs(self):
         self.assertEqual(protlib.protfunc_parser(
             "$eval({'test': '1', 2:3, 3: $toint(3.5)})"), {'test': '1', 2: 3, 3: 3})
 
+        self.assertEqual(protlib.protfunc_parser("obj(#1)", session=self.session), 'obj(#1)')
         self.assertEqual(protlib.protfunc_parser("$obj(#1)", session=self.session), '#1')
+        self.assertEqual(protlib.protfunc_parser("dbref(#1)", session=self.session), 'dbref(#1)')
+        self.assertEqual(protlib.protfunc_parser("$dbref(#1)", session=self.session), '#1')
+        self.assertEqual(protlib.protfunc_parser("$badfunc(#1)", session=self.session), '<UNKNOWN>')
+        self.assertEqual(protlib.protfunc_parser("stone(#12345)", session=self.session), 'stone(#12345)')
         self.assertEqual(protlib.protfunc_parser("#1", session=self.session), '#1')
+        self.assertEqual(protlib.protfunc_parser("#12345", session=self.session), '#12345')
+        self.assertEqual(protlib.protfunc_parser("nothing(#1)", session=self.session), 'nothing(#1)')
+        self.assertEqual(protlib.protfunc_parser("(#12345)", session=self.session), '(#12345)')
         self.assertEqual(protlib.protfunc_parser("$obj(Char)", session=self.session), '#6')
-        self.assertEqual(protlib.protfunc_parser("$obj(Char)", session=self.session), '#6')
+        self.assertEqual(protlib.protfunc_parser("obj(Char)", session=self.session), 'obj(Char)')
         self.assertEqual(protlib.protfunc_parser("$objlist(#1)", session=self.session), ['#1'])
+        self.assertEqual(protlib.protfunc_parser("objlist(#1)", session=self.session), 'objlist(#1)')
+        self.assertEqual(protlib.protfunc_parser("dbref(Char)", session=self.session), 'dbref(Char)')
+        self.assertEqual(protlib.protfunc_parser("$dbref(Char)", session=self.session), '<UNKNOWN>')
```

Comment:
This last test proves that `$dbref()` is expected to have a valid `#dbref` as passed in param.

Reply:
While passing, it's unfortunately passing for the wrong reason since this is not added as a proper protfunc.

Reply:
It's passing in the sense that the expected result is not the successful search **but** `"<UNKNOWN>"` which seems to be the keyword to signify failure executing.